model/src/order.rs

```diff
@@ -41,6 +42,15 @@ impl Default for Order {
     }
 }
 
+#[derive(Eq, PartialEq, Clone, Debug, Deserialize, Serialize, Hash)]
+#[serde(rename_all = "camelCase")]
+pub enum OrderStatus {
+    Open,
```

Comment:
I suppose partially filled falls back to a status of Open. What about partially filled orders that are no longer open? I think we need a `PartiallyFilled` status.

Reply:
> Seems alright, but I think there are a few cases missing in the test. All of my concerns lie with the logic for partially fillable orders. At this moment the fallback is not terrible... partially filled order that is still open and not canceled is "Open" and when it is expired its expired. I just think that some relevant information for the front end is lost.
> 
> I know that the FE doesn't currently support partial fulfillment, but if we don't incorporate this now, it will be forgotten and eventually be reported as a bug.

Nick raised the same point and I have replied in the comment https://github.com/gnosis/gp-v2-services/pull/714#discussion_r650116518

We what are your thoughts based on that?

Reply:
> I suppose partially filled falls back to a status of Open. 

Yes. It's `Open` as long as it's not fully filled.

> What about partially filled orders that are no longer open? I think we need a PartiallyFilled status.

You can take how this is handled on the explorer as an example:
https://github.com/gnosis/gp-ui/blob/ad8223cc234b5f2592176cd0d2b39afdb86e6d4c/src/utils/operator.ts#L249-L251

The order status is independent of the fill status, and there's some duplication on the flags `partialFilled` and `fullyFilled`.

Also, the interface might want to focus on the `PartiallyFilled` status, or it might want to know what's the order status.
Because we can have and order as `Open`, `Cancelled` and `Expired` being `PartiallyFilled`.